services/src/main/java/org/keycloak/broker/saml/SAMLIdentityProvider.java

```diff
@@ -162,6 +167,12 @@ public void backchannelLogout(KeycloakSession session, UserSessionModel userSess
         JaxrsSAML2BindingBuilder binding = buildLogoutBinding(session, userSession, realm);
         try {
             LogoutRequestType logoutRequest = buildLogoutRequest(userSession, uriInfo, realm, singleLogoutServiceUrl);
+            if (logoutRequest.getDestination() != null) {
+                singleLogoutServiceUrl = logoutRequest.getDestination().toString();
+            } else {
```

Comment:
Please do remove the `else` part. In case destination is not set, the original value of `singleLogoutServiceUrl` should be used for submission.

Reply:
I managed this case exactly as it is done a few lines above. This method does not support blank values for `singleLogoutServiceUrl`, should we allow blank values after building the logout request anyway?

https://github.com/keycloak/keycloak/blob/f8a4f66d6c183de894a8a436f9e35dfa3248e862/services/src/main/java/org/keycloak/broker/saml/SAMLIdentityProvider.java#L163

Reply:
By this line, the `singleLogoutServiceUrl` is guaranteed to be non-null and non-empty exactly because of the mentioned check. Hence use the value from the `singleLogoutServiceUrl`.

Reply:
Based on [this comment](https://github.com/keycloak/keycloak/pull/7007#discussion_r452036172), I understood that you wanted me to put the request's destination in`singleLogoutServiceUrl` because custom processors could be involved in `buildLogoutRequest`  (as we do for the destination url in the login flow). I can remove this, no problem.